### PR TITLE
Add more feedback for users of /override

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Two of the most common issues we see after rolling out the override
plugin are that admins will not specify a context to override, or they
will specify the "common name" of the context that they would use to
trigger the job in the first place. We should give them feedback for
these cases.

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

/assign @cjwagner @fejta 
/cc @BenTheElder @krzyzacy

### CODE REVIEW

File: prow/plugins/override/override.go [hunk 1]
```diff
@@ -147,20 +155,25 @@ func handle(oc overrideClient, log *logrus.Entry, e *github.GenericCommentEvent)
 
 	mat := overrideRe.FindAllStringSubmatch(e.Body, -1)
 	if len(mat) == 0 {
+		// no /override commands given in the comment
 		return nil
```
Comment (fejta) [inline prow/plugins/override/override.go]: nit, I'd inline this with the return

File: prow/plugins/override/override.go [hunk 2]
```diff
@@ -147,20 +155,25 @@ func handle(oc overrideClient, log *logrus.Entry, e *github.GenericCommentEvent)
...
-	for _, m := range mat {
-		overrides.Insert(m[1])
-	}
-
 	org := e.Repo.Owner.Login
 	repo := e.Repo.Name
 	number := e.Number
 	user := e.User.Login
 
+	overrides := sets.String{}
```
Comment (fejta) [inline prow/plugins/override/override.go]: nit: please use `var overrides sets.String` when you're setting a default value

File: prow/plugins/override/override.go [hunk 3]
```diff
@@ -182,6 +195,24 @@ func handle(oc overrideClient, log *logrus.Entry, e *github.GenericCommentEvent)
 		return oc.CreateComment(org, repo, number, plugins.FormatResponseRaw(e.Body, e.HTMLURL, user, resp))
 	}
 
+	contexts := sets.NewString()
```
Comment (fejta) [inline prow/plugins/override/override.go]: nit: Please use a consistent pattern

File: prow/plugins/override/override.go (hunk 2 above)
Comment (stevekuznetsov) [reply prow/plugins/override/override.go]: Oops, this should have been `NewString()` (I think they want us to call the constructor?)

**Comment by stevekuznetsov** [comment]
/hold cancel
@fejta updated

File: prow/plugins/override/override.go [hunk 4]
```diff
@@ -37,7 +37,7 @@ import (
 const pluginName = "override"
 
 var (
-	overrideRe = regexp.MustCompile(`(?mi)^/override (.+?)\s*$`)
+	overrideRe = regexp.MustCompile(`(?mi)^/override( (.+?)\s*)?$`)
```
Comment (cjwagner) [inline prow/plugins/override/override.go]: This doesn't need to be a capturing group. Adding an additional capturing group to check if the value is empty doesn't provide any more value than using the existing capture group and making this new group a non-capturing group.

File: prow/plugins/override/override.go [hunk 5]
```diff
@@ -147,20 +155,24 @@ func handle(oc overrideClient, log *logrus.Entry, e *github.GenericCommentEvent)
...
 	org := e.Repo.Owner.Login
 	repo := e.Repo.Name
 	number := e.Number
 	user := e.User.Login
 
+	overrides := sets.NewString()
+	for _, m := range mat {
+		if m[1] == "" {
+			resp := "/override requires a failed status context to operate on, but none was given"
+			log.Warn(resp)
```
Comment (cjwagner) [inline prow/plugins/override/override.go]: Can we switch this warning (and the one below) to debug level? There isn't any unexpected behavior here, everything is WAI, this is entirely user error and doesn't require attention from a Prow admin.

**Pull request merged**